src/ops/fixit.rs

```diff
@@ -56,18 +61,60 @@ fn exec(args: FixitArgs) -> CargoResult<()> {
     let mut iteration = 0;
 
     let mut last_errors = IndexMap::new();
-    let mut current_target: Option<BuildUnit> = None;
+    let mut current_target: Option<(BuildUnit, IndexSet<String>)> = None;
```

Comment:
Instead of tracking the original error, what if revert and then re-run? Broken code shouldn't come up too often and the perf hit shouldn't matter too much.

Reply:
That doesn't sound too bad